### ISSUE TITLE
Lack of documentation. Why ?

### ISSUE THREAD

**Issue by ErrorxCode** [instagram4j/instagram4j#638]
### Pre-Checklist

I made sure...
- [✔ ] I read the README.md
- [ ✔] I looked over the examples and wiki
- [✔ ] my issue hasn't been asked or solved before

---

### Describe your feature request

Why this project has no documentation? How would users know what method actually does?
There are a wide no. of issues because there is no documentation. These are not in fact problems but the tutorial or guide for doing something. I request you to please create a javadoc that tells about at least what the method does.

Thank you.

**Comment by mikebgrep** [comment]
I vouch for this also.
Hope it will be done it's needed to use the api.
Thanks

**Comment by ErrorxCode** [comment]
> 

Well, for now, you can use [EasyInsta](https://github.com/ErrorxCode/EasyInsta). This library is the wrapper of this API and covers almost all of the common methods.

**Comment by jvogit** [comment]
I agree the documentation and the actual API wrapper needs a lot of work. Unfortunately, I don't have a lot of time to contribute currently as I'm finishing up my studies in university. I think I'll come to writing some javadocs eventually when I have some time.

instagram4j does have a limited wrapper actually called ["Actions"](https://github.com/instagram4j/instagram4j/tree/master/src/main/java/com/github/instagram4j/instagram4j/actions) that serves to abstract certain functionality like uploading a photo/video on a timeline. And yes it needs a lot of work to implement more features and of course documentation.

I apologize if it isn't the answer you guys are looking for from me 😅 I try my best to fit in as much as I can but life happens!

**Comment by ErrorxCode** [comment]
> I agree the documentation and the actual API wrapper need a lot of work. Unfortunately, I don't have a lot of time to contribute currently as I'm finishing up my studies at university. I think I'll come to writing some javadocs eventually when I have some time.
> 
> instagram4j does have a limited wrapper actually called ["Actions"](https://github.com/instagram4j/instagram4j/tree/master/src/main/java/com/github/instagram4j/instagram4j/actions) that serves to abstract certain functionality like uploading a photo/video on a timeline. And yes it needs a lot of work to implement more features and of course documentation.
> 
> I apologize if it isn't the answer you guys are looking for from me 😅 I try my best to fit in as much as I can but life happens! And it so happens I'm forced to put this on the backburner for a long time.

I can understand. I am also a student and have to manage time for these things. I could collaborate in writing documentation, but I myself don't know what every method does. Feel free to [contact](https://www.instagram.com/x__coder__x/) me in future.
